uniffi_bindgen/src/bindings/gecko/templates/SharedHeaderTemplate.h

```diff
@@ -0,0 +1,794 @@
+/// `ViaFfi<T, U>::Lift(U, T)` compiles, we know that a value of type `U` from
+/// the FFI can be lifted into a value of type `T`.
+template <typename T, typename FfiType>
+struct ViaFfi {
+  static bool Lift(const FfiType& aLowered, T& aLifted) = delete;
+  static FfiType Lower(const T& aLifted) = delete;
+};
+
+// This macro generates boilerplate specializations for primitive numeric types
+// that are passed directly over the FFI without conversion.
+#define UNIFFI_SPECIALIZE_SERIALIZABLE_PRIMITIVE(Type, readFunc, writeFunc) \
+  template <>                                                               \
+  struct Serializable<Type> {                                               \
+    static CheckedInt<size_t> Size(const Type& aValue) {                    \
+      return sizeof(Type);                                                  \
+    }                                                                       \
+    static MOZ_MUST_USE bool ReadFrom(Reader& aReader, Type& aValue) {      \
+      aValue = aReader.readFunc();                                          \
+      return true;                                                          \
+    }                                                                       \
+    static void WriteInto(Writer& aWriter, const Type& aValue) {            \
+      aWriter.writeFunc(aValue);                                            \
+    }                                                                       \
+  };                                                                        \
+  template <>                                                               \
+  struct ViaFfi<Type, Type> {                                               \
+    static MOZ_MUST_USE bool Lift(const Type& aLowered, Type& aLifted) {    \
+      aLifted = aLowered;                                                   \
+      return true;                                                          \
+    }                                                                       \
+    static MOZ_MUST_USE Type Lower(const Type& aLifted) { return aLifted; } \
+  }
+
+UNIFFI_SPECIALIZE_SERIALIZABLE_PRIMITIVE(uint8_t, ReadUInt8, WriteUInt8);
+UNIFFI_SPECIALIZE_SERIALIZABLE_PRIMITIVE(int8_t, ReadInt8, WriteInt8);
+UNIFFI_SPECIALIZE_SERIALIZABLE_PRIMITIVE(uint16_t, ReadUInt16, WriteUInt16);
+UNIFFI_SPECIALIZE_SERIALIZABLE_PRIMITIVE(int16_t, ReadInt16, WriteInt16);
+UNIFFI_SPECIALIZE_SERIALIZABLE_PRIMITIVE(uint32_t, ReadUInt32, WriteUInt32);
+UNIFFI_SPECIALIZE_SERIALIZABLE_PRIMITIVE(int32_t, ReadInt32, WriteInt32);
+UNIFFI_SPECIALIZE_SERIALIZABLE_PRIMITIVE(uint64_t, ReadUInt64, WriteUInt64);
+UNIFFI_SPECIALIZE_SERIALIZABLE_PRIMITIVE(int64_t, ReadInt64, WriteInt64);
+UNIFFI_SPECIALIZE_SERIALIZABLE_PRIMITIVE(float, ReadFloat, WriteFloat);
+UNIFFI_SPECIALIZE_SERIALIZABLE_PRIMITIVE(double, ReadDouble, WriteDouble);
+
+/// Booleans are passed as unsigned integers over the FFI, because JNA doesn't
```

Comment:
Change "unsigned integers" to "Uint8"s... I was briefly confused because I erroneously assumed a UInt32.

Reply:
Ahh, oops, good call! Done! I also called out that it's an implementation detail of UniFFI, instead of mentioning JNA specifically.